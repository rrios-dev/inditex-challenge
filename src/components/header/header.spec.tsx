import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import FavsProvider from "@/pods/favs/favs-provider";

import Header from "./header";

vi.mock("next/navigation", async (importOriginal) => ({
  ...((await importOriginal()) as any),
  useRouter: () => {},
}));

describe("Header", () => {
  it.each(["row", "column"] as const)("%s", (direction) => {
    const { asFragment } = render(
      <FavsProvider>
        <Header />
      </FavsProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
