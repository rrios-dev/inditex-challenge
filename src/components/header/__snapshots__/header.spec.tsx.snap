// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > column 1`] = `
<DocumentFragment>
  <header
    class="_header_98091d"
  >
    <div
      class="_container_98091d"
    >
      <a
        aria-label="Go to hero finder"
        class="_link_dbb513"
        href="/"
      >
        <img
          alt="Marvel logo"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="52"
          src="/img/marvel-logo.svg"
          style="color: transparent;"
          width="130"
        />
      </a>
      <a
        aria-label="Go to favs hero list"
        class="_link_dbb513 _header-heart-link_3392a9"
        href="/favs"
      >
        <img
          alt="Empty heart"
          data-nimg="1"
          decoding="async"
          height="22"
          loading="lazy"
          src="/icons/heart-empty.svg"
          style="color: transparent;"
          width="24"
        />
        0
      </a>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Header > row 1`] = `
<DocumentFragment>
  <header
    class="_header_98091d"
  >
    <div
      class="_container_98091d"
    >
      <a
        aria-label="Go to hero finder"
        class="_link_dbb513"
        href="/"
      >
        <img
          alt="Marvel logo"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="52"
          src="/img/marvel-logo.svg"
          style="color: transparent;"
          width="130"
        />
      </a>
      <a
        aria-label="Go to favs hero list"
        class="_link_dbb513 _header-heart-link_3392a9"
        href="/favs"
      >
        <img
          alt="Empty heart"
          data-nimg="1"
          decoding="async"
          height="22"
          loading="lazy"
          src="/icons/heart-empty.svg"
          style="color: transparent;"
          width="24"
        />
        0
      </a>
    </div>
  </header>
</DocumentFragment>
`;
